### PR TITLE
Some big improvements

### PR DESCRIPTION
**Pull request by xMasterX**
## Modifications made
- Ability to change baudrate using Up button, hold button to switch between baudrates (9600, 19200, 38400, 57600, 115200) (i set 57600 as default)
- Ok button will set backlight to always on mode, to disable press ok button again (it will restore default settings after app exit too)
- Long press Right button to change speed from knots to kilometers per hour
- Exit from app using long press on back button instead of short press, may be useful in case you want to turn backlight on and accidentally click back

+ ./fbt format - was applied
+ Stability fixes

### CODE REVIEW

**Comment by ezod** [comment]
Awesome, thank you!

Any particular reason for 57600 as default baud rate? It looks to me like most modules output at 9600.

I also don't want to apply the `./fbt format` to `minmea.h` or `minmea.c` as those are third party, can we remove that from the diff? Ideally in general, apply big formatting changes in a separate PR, or at least a separate commit, as it makes it difficult to review the functional changes.

File: gps_uart.h [hunk 1]
```diff
@@ -3,37 +3,44 @@
...
-  float course;
-  float altitude;
-  char altitude_units;
-  int fix_quality;
-  int satellites_tracked;
-  int time_hours;
-  int time_minutes;
-  int time_seconds;
+static const int gps_baudrates[5] = {9600, 19200, 38400, 57600, 115200};
+static int current_gps_baudrate = 3;
```
Comment (ezod) [inline gps_uart.h]:
```suggestion
static int current_gps_baudrate = 0;
```

**Comment by xMasterX** [comment]
> Awesome, thank you!
> 
> Any particular reason for 57600 as default baud rate? It looks to me like most modules output at 9600.
> 
> I also don't want to apply the `./fbt format` to `minmea.h` or `minmea.c` as those are third party, can we remove that from the diff? Ideally in general, apply big formatting changes in a separate PR, or at least a separate commit, as it makes it difficult to review the functional changes.

I don't have much time to slice that into two PRs
You can apply fbt format to your current source then you will see only actual changes without formatter things
About default value - i set it to 57k since my module uses that, and it was making testing easier in my case
You can change it to value that you think will fit the most

And about `minmea.h` and `minmea.c`
I can revet format changes from them

**Comment by ezod** [comment]
This is ready to merge once the changes to `minmea.h` and `minmea.c` are reverted. I don't see an easy way to do that from the web UI, but locally you can either make a revert commit, or use `amend` or `rebase` better yet.

**Pull request merged**